pkg/webhook/preflight/nutanix/storagecontainer.go

```diff
@@ -145,20 +145,35 @@ func newStorageContainerChecks(cd *checkDependencies) []preflight.Check {
 		return checks
 	}
 
-	if cd.nutanixClusterConfigSpec != nil && cd.nutanixClusterConfigSpec.ControlPlane != nil &&
+	if cd.nutanixClusterConfigSpec != nil &&
+		cd.nutanixClusterConfigSpec.ControlPlane != nil &&
 		cd.nutanixClusterConfigSpec.ControlPlane.Nutanix != nil {
-		checks = append(checks,
-			&storageContainerCheck{
-				machineSpec: &cd.nutanixClusterConfigSpec.ControlPlane.Nutanix.MachineDetails,
-				field:       "$.spec.topology.variables[?@.name==\"clusterConfig\"].value.controlPlane.nutanix.machineDetails",
-				csiSpec:     &cd.nutanixClusterConfigSpec.Addons.CSI.Providers.NutanixCSI,
-				nclient:     cd.nclient,
-			},
-		)
+		// Skip the check if failureDomains are configured
+		if len(cd.nutanixClusterConfigSpec.ControlPlane.Nutanix.FailureDomains) > 0 {
+			cd.log.Info("Skip preflight check NutanixStorageContainer for controlPlane with failureDomains configured.")
```

Comment:
Other checks do not log when they skip. If we want to log this at all, we should only do so when verbosity is increased.